app/views/discovered_hosts/show.html.erb

```diff
@@ -2,15 +2,55 @@
 
 <%= discovered_hosts_title_actions(@host) %>
 
-<h3><%= _('Facts discovered on this host') -%></h3>
-<table class="table table-bordered table-striped table-condensed">
-  <tr>
-    <th><%= _('Fact') -%></th>
-    <th><%= _('Value') -%></th>
-  <% @host.facts_hash.sort.each do |fact| -%>
-    <tr>
-      <td><%= fact[0] %></td>
-      <td><%= truncate(fact[1], :length => 100) %></td>
-    </tr>
+<div class="row">
+  <% @general.each do |keys, val| %>
+    <div class="col-md-6">
+      <div class="panel panel-default">
+        <div class="panel-heading" ><strong> <%= keys %></strong> </div>
+          <div class="panel-body">
+            <ul class="list-group facts-list">
+              <% @facts.sort.select{|key, value| val.any?{|v| (key.include?(v) &&  !key.include?("bios") && !key.include?("ipmi")) || (key.eql? "ipaddress")}}.each do |fact| %>
+                <li class="list-group-item"> <strong> <%= fact[0] %>: </strong>  <%= truncate(fact[1], :length => 50)%></li>
+                <% @facts.delete(fact[0]) %>
+              <% end %>
```

Comment:
Ok this is code logic in view template, not good. Can you move all the code logic in the controller? Pass the data as hash of hashes and in views just blindly render them.